encoding.go

```diff
@@ -131,6 +131,22 @@ func newBuffer(data []byte) *byteBuffer {
 	}
 }
 
+func newZeroPaddedBuffer(data []byte, length int) *byteBuffer {
+	if len(data) >= length {
+		return newBuffer(data)
+	}
+
+	padLength := length - len(data)
+	paddedData := make([]byte, length)
+	for i := range paddedData {
+		paddedData[i] = 0
+		if i >= padLength {
+			paddedData[i] = data[i-padLength]
+		}
+	}
+	return newBuffer(paddedData)
```

Comment:
Explicit array initialization shouldn't be necessary, see http://blog.golang.org/go-slices-usage-and-internals.

I believe the following block of code is sufficient:

	if len(data) >= length {
		return newBuffer(data)
	}
	pad := make([]byte, length - len(data))
	return newBuffer(append(pad, data...))